firewood/src/merkle.rs

```diff
@@ -1182,6 +1186,20 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
     }
 }
 
+fn set_parent(new_chd: DiskAddress, parents: &mut [(ObjRef<'_, Node>, u8)]) {
+    let (p_ref, idx) = parents.last_mut().unwrap();
+    p_ref
+        .write(|p| {
+            match &mut p.inner {
+                NodeType::Branch(pp) => pp.chd[*idx as usize] = Some(new_chd),
+                NodeType::Extension(pp) => *pp.chd_mut() = new_chd,
+                _ => unreachable!(),
+            }
+            p.rehash();
+        })
+        .unwrap();
+}
```

Comment:
No changes from https://github.com/ava-labs/firewood/pull/311/files#diff-6a6e300f3d9050248dbafa719531f14eabecc1bfac2169bb2527c81ee5474ecaL159 other than removing `self` reference that was unused.